gobblin-config-management/gobblin-config-core/src/main/java/gobblin/config/store/hdfs/SimpleHDFSConfigStore.java

```diff
@@ -0,0 +1,370 @@
+ */
+public class SimpleHDFSConfigStore implements ConfigStore {
+
+  public static final String CONFIG_STORE_NAME = "_CONFIG_STORE";
+
+  private final FileSystem fs;
+  private final URI physicalStoreRoot;
+  private final URI logicalStoreRoot;
+  private final Cache<String, Path> versions;
+
+  /**
+   * Constructs a {@link SimpleHDFSConfigStore} using a given {@link FileSystem} and a {@link URI} that points to the
+   * physical location of the store root.
+   *
+   * @param fs the {@link FileSystem} the {@link ConfigStore} is stored on.
+   * @param physicalStoreRoot the fully qualified {@link URI} of the physical store root, the {@link URI#getScheme()} of the
+   *                          {@link URI} should match the {@link FileSystem#getScheme()} of the given {@link FileSystem}.
+   * @param logicalStoreRoot the fully qualfied {@link URI} of the logical store root
+   */
+  protected SimpleHDFSConfigStore(FileSystem fs, URI physicalStoreRoot, URI logicalStoreRoot) {
+    Preconditions.checkNotNull(fs, "fs cannot be null!");
+    Preconditions.checkNotNull(physicalStoreRoot, "physicalStoreRoot cannot be null!");
+    Preconditions.checkNotNull(logicalStoreRoot, "logicalStoreRoot cannot be null!");
+
+    this.fs = fs;
+
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(physicalStoreRoot.getScheme()),
+        "The physicalStoreRoot must have a valid scheme!");
+    Preconditions.checkArgument(physicalStoreRoot.getScheme().equals(fs.getUri().getScheme()),
+        "The scheme of the physicalStoreRoot and the filesystem must match!");
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(physicalStoreRoot.getPath()),
+        "The path of the physicalStoreRoot must be valid as it is the root of the store!");
+
+    this.physicalStoreRoot = physicalStoreRoot;
+    this.logicalStoreRoot = logicalStoreRoot;
+    this.versions = CacheBuilder.newBuilder().build();
+  }
+
+  /**
+   * Returns a {@link String} representation of the highest version stored in the {@link ConfigStore}. This method
+   * determines the highest version by doing an {@code ls} on the store root, and sorting the output based on the
+   * {@link FileStatus#getPath()}.
+   *
+   * @return a {@link String} representing the current (highest) version of the {@link ConfigStore}.
+   */
+  @Override
+  public String getCurrentVersion() {
+    Path configStoreDir = new Path(new Path(this.physicalStoreRoot), CONFIG_STORE_NAME);
+
+    try {
+      if (!this.fs.exists(configStoreDir)) {
+        throw new VersionDoesNotExistException(getStoreURI(), "CURRENT", "");
+      }
+    } catch (IOException e) {
+      throw new RuntimeException(
+          String.format("Error while checking if the configStoreDir: \"%s\" exists", configStoreDir), e);
+    }
+
+    try {
+      FileStatus[] fileStatuses = this.fs.listStatus(configStoreDir);
+      Arrays.sort(fileStatuses, new FileStatusPathNameComparator());
+      return fileStatuses[fileStatuses.length - 1].getPath().getName();
```

Comment:
What do you mean exactly? The method currently returns the version name, e.g. `v1.0` or `v1.1`. Do you want the full path to the version?